skbio/metadata/_testing.py

```diff
@@ -904,3 +905,245 @@ def test_has_positional_metadata(self):
+        self.assertIsNot(obj.interval_metadata._intervals,
+                         obj_copy.interval_metadata._intervals)
+        for i, j in zip(obj.interval_metadata._intervals,
+                        obj_copy.interval_metadata._intervals):
+            self.assertIsNot(i, j)
+            self.assertIsNot(i.metadata, j.metadata)
+            self.assertIsNot(i.metadata['gene'], j.metadata['gene'])
+            self.assertIs(i.metadata['product'], j.metadata['product'])
+
+    def test_deepcopy_interval_metadata_empty(self):
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        obj_copy = copy.deepcopy(obj)
+
+        self.assertEqual(obj, obj_copy)
+        self.assertIsNot(obj, obj_copy)
+
+    def test_deepcopy_memo_is_respected(self):
+        # Basic test to ensure deepcopy's memo is passed through to recursive
+        # deepcopy calls.
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        memo = {}
+        copy.deepcopy(obj, memo)
+        self.assertGreater(len(memo), 1)
+
+    def test_interval_metadata_getter(self):
+        self.im.add(**self.intvls[0])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        self.assertIsInstance(obj.interval_metadata, IntervalMetadata)
+        self.assertEqual(self.im, obj.interval_metadata)
+
+        # Update existing column.
+        obj.interval_metadata._intervals[0].metadata['gene'] = 'sagB'
+        self.assertNotEqual(obj.interval_metadata, self.im)
+        self.im._intervals[0].metadata['gene'] = 'sagB'
+        self.assertEqual(obj.interval_metadata, self.im)
+
+        # Add new interval feature.
+        obj.interval_metadata.add(**self.intvls[1])
+        self.im.add(**self.intvls[1])
+        self.assertEqual(obj.interval_metadata, self.im)
+
+    def test_interval_metadata_setter_makes_copy(self):
+        intvl = self.im.add(**self.intvls[1])
+        obj = self._interval_metadata_constructor_(self.upper_bound)
+        obj.interval_metadata = self.im
+
+        self.assertEqual(obj.interval_metadata, self.im)
+        self.assertIsNot(obj.interval_metadata, self.im)
+
+        # Changing mutable value of metadata of the old interval
+        # also changes obj.
+        intvl.metadata['gene'].append('b')
+        self.assertEqual(obj.interval_metadata, self.im)
+
+        # Changing old interval doesn't change obj
+        intvl.locations = [(3, 6)]
+        self.assertNotEqual(obj.interval_metadata, self.im)
+
+    def test_interval_metadata_setter_invalid_type(self):
+        self.im.add(**self.intvls[0])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+
+        for i in [2, None, '', {}, []]:
+            with self.assertRaisesRegex(
+                    TypeError,
+                    'You must provide `IntervalMetadata` object'):
+                obj.interval_metadata = i
+
+        self.assertEqual(self.im, obj.interval_metadata)
+
```

Comment:
Can you add a test case similar to `test_positional_metadata_setter_len_mismatch`?